### ISSUE TITLE
weave-proxy start failure with docker 1.12

### ISSUE THREAD

**Issue by osallou** [weaveworks/weave#2483]
After executing weave launch, weave proxy fails and network is not seen in *docker networks*.

Logs of weave-proxy container:

    INFO: 2016/08/17 14:58:06.973283 weave proxy 1.6.0
    INFO: 2016/08/17 14:58:06.973314 Command line arguments: -H unix:///var/run/weave/weave.sock
    INFO: 2016/08/17 14:58:06.974220 Docker API on unix:///var/run/docker.sock: &[Version=1.12.0 ApiVersion=1.24 GitCommit=8eab29e GoVersion=go1.6.3 Os=linux Arch=amd64 KernelVersion=4.2.0-42-generic BuildTime=2016-07-28T22:05:53.080079366+00:00]
    FATA: 2016/08/17 14:58:06.975608 Could not start proxy: Could not find the weavewait volume

Docker info:

    Server Version: 1.12.0
    Storage Driver: aufs
     Root Dir: /var/lib/docker/aufs
     Backing Filesystem: extfs
     Dirs: 789
     Dirperm1 Supported: true
    Logging Driver: json-file
    Cgroup Driver: cgroupfs
    Plugins:
     Volume: local
     Network: null overlay host bridge
    Swarm: inactive
    Runtimes: runc
    Default Runtime: runc
    Security Options: apparmor seccomp
    Kernel Version: 4.2.0-42-generic
    Operating System: Ubuntu 15.10

**Comment by osallou** [comment]
Here are weave-proxy mounts:

        "Mounts": [
            {
                "Source": "/var/run/docker.sock",
                "Destination": "/var/run/docker.sock",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Source": "/var/run/weave",
                "Destination": "/var/run/weave",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ]

**Comment by awh** [comment]
Hi @osallou 

Does `weave reset` followed by `weave launch` fix the issue?

**Comment by osallou** [comment]
after a reset and a launch, I only have 

    osallou@tifenn~/Development/NOSAVE/go-docker/weave $ sudo docker ps 
    CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS               NAMES
    f9167709c5d9        weaveworks/weave:1.6.0   "/home/weave/weaver -"   58 seconds ago      Up 55 seconds

**Comment by osallou** [comment]
after a reset and setup, launch still fails.
If however I run 

    weave launch-router
   weave launch-plugin

it seems to work, network appears in **docker network list**

**Comment by brb** [comment]
@osallou Could you run `weave reset && WEAVE_DEBUG=1 weave launch` and provide an output? Also, after running the cmds, could you do `docker inspect weavevolumes-1.6.0` and include its output?

**Comment by osallou** [comment]
Here is result.
No docker container is running with reset :-(
If I run with a reset --force (as no weave process was running)

    + SCRIPT_VERSION=1.6.0
    + IMAGE_VERSION=latest
    + [ 1.6.0 = (unreleased version) ]
    + IMAGE_VERSION=1.6.0
    + IMAGE_VERSION=1.6.0
    + MIN_DOCKER_VERSION=1.6.0
    + DOCKERHUB_USER=weaveworks
    + BASE_EXEC_IMAGE=weaveworks/weaveexec
    + EXEC_IMAGE=weaveworks/weaveexec:1.6.0
    + WEAVEDB_IMAGE=weaveworks/weavedb
    + + cut -s -d: -f1
    echo 
    + PROXY_HOST=
    + PROXY_HOST=127.0.0.1
    + DOCKER_CLIENT_HOST=
    + IP_REGEXP=[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}
    + CIDR_REGEXP=[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/[0-9]{1,2}
    + check_docker_version
    + + seddocker -n -e -v s|^Docker version \([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*|\1|p

    + DOCKER_VERSION=1.12.0
    + [ -z 1.12.0 ]
    + version_lt 1.12.0 1.6.0
    + VERSION_MAJOR=1
    + REST=1.12 VERSION_MINOR=12
    + VERSION_PATCH=0
    + MIN_VERSION_MAJOR=1
    + REST=1.6 MIN_VERSION_MINOR=6
    + MIN_VERSION_PATCH=0
    + [ ( 1 -lt 1 ) -o ( 1 -eq 1 -a ( 12 -lt 6 -o ( 12 -eq 6 -a ( 0 -lt 0 ) ) ) ) ]
    + return 1
    + [ launch = --local ]
    + handle_help_arg launch
    + [ launch = --help ]
    + [ launch = version -a -z  ]
    + [ launch = env -a  = --restore ]
    + [ launch = run -a -z  ]
    + [ -z  ]
    + exec_remote launch
    + docker_run_options
    + docker_sock_options
    + echo 
    + grep -q ^unix://
    + echo -v /var/run/docker.sock:/var/run/docker.sock
    + echo --privileged --net=host -v /var/run/docker.sock:/var/run/docker.sock
    + cni_volume_options launch
    + [ launch = setup -o launch = setup-cni ]
    + launch_router_options launch
    + [ launch = launch -o launch = launch-router ]
    + echo -v /:/host -e HOST_ROOT=/host
    + docker run --rm --privileged --net=host -v /var/run/docker.sock:/var/run/docker.sock --pid=host -v /:/host -e HOST_ROOT=/host -e DOCKERHUB_USER=weaveworks -e WEAVE_VERSION -e WEAVE_DEBUG -e WEAVE_DOCKER_ARGS -e WEAVEPROXY_DOCKER_ARGS -e WEAVEPLUGIN_DOCKER_ARGS -e WEAVE_PASSWORD -e WEAVE_PORT -e WEAVE_HTTP_ADDR -e WEAVE_CONTAINER_NAME -e WEAVE_MTU -e WEAVE_NO_FASTDP -e WEAVE_NO_BRIDGED_FASTDP -e WEAVE_NO_PLUGIN -e DOCKER_BRIDGE -e DOCKER_CLIENT_HOST= -e DOCKER_CLIENT_ARGS -e PROXY_HOST=127.0.0.1 -e COVERAGE -e CHECKPOINT_DISABLE -e AWSVPC weaveworks/weaveexec:1.6.0 --local launch
    + SCRIPT_VERSION=1.6.0
    + IMAGE_VERSION=latest
    + [ 1.6.0 = (unreleased version) ]
    + IMAGE_VERSION=1.6.0
    + IMAGE_VERSION=1.6.0
    + MIN_DOCKER_VERSION=1.6.0
    + DOCKERHUB_USER=weaveworks
    + BASE_EXEC_IMAGE=weaveworks/weaveexec
    + EXEC_IMAGE=weaveworks/weaveexec:1.6.0
    + WEAVEDB_IMAGE=weaveworks/weavedb
    + PROXY_HOST=127.0.0.1
    + PROXY_HOST=127.0.0.1
    + DOCKER_CLIENT_HOST=
    + IP_REGEXP=[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}
    + CIDR_REGEXP=[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/[0-9]{1,2}
    + check_docker_version
    + docker -v
    + sed -n -e s|^Docker version \([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*|\1|p
    + DOCKER_VERSION=1.6.2
    + [ -z 1.6.2 ]
    + version_lt 1.6.2 1.6.0
    + VERSION_MAJOR=1
    + REST=1.6 VERSION_MINOR=6
    + VERSION_PATCH=2
    + MIN_VERSION_MAJOR=1
    + REST=1.6 MIN_VERSION_MINOR=6
    + MIN_VERSION_PATCH=0
    + [ ( 1 -lt 1 ) -o ( 1 -eq 1 -a ( 6 -lt 6 -o ( 6 -eq 6 -a ( 2 -lt 0 ) ) ) ) ]
    + return 1
    + [ --local = --local ]
    + shift 1
    + IS_LOCAL=1
    + handle_help_arg launch
    + [ launch = --help ]
    + [ launch = version -a -z 1 ]
    + [ launch = env -a  = --restore ]
    + [ launch = run -a -z 1 ]
    + [ -z 1 ]
    + RESTART_POLICY=--restart=always
    + BASE_IMAGE=weaveworks/weave
    + IMAGE=weaveworks/weave:1.6.0
    + CONTAINER_NAME=weave
    + BASE_PLUGIN_IMAGE=weaveworks/plugin
    + PLUGIN_IMAGE=weaveworks/plugin:1.6.0
    + PLUGIN_CONTAINER_NAME=weaveplugin
    + VOLUMES_CONTAINER_NAME=weavevolumes-1.6.0
    + DB_CONTAINER_NAME=weavedb
    + DOCKER_BRIDGE=docker0
    + BRIDGE=weave
    + DATAPATH=datapath
    + CONTAINER_IFNAME=ethwe
    + BRIDGE_IFNAME=vethwe-bridge
    + DATAPATH_IFNAME=vethwe-datapath
    + PCAP_IFNAME=vethwe-pcap
    + PORT=6783
    + HTTP_ADDR=127.0.0.1:6784
    + PROXY_PORT=12375
    + PROXY_CONTAINER_NAME=weaveproxy
    + COVERAGE_ARGS=
    + [ -n  ]
    + id -u
    + [ 0 = 0 ]
    + uname -s -r
    + sed -n -e s|^\([^ ]*\) \([0-9][0-9]*\)\.\([0-9][0-9]*\).*|\1 \2 \3|p
    + read sys maj min
    + [ Linux != Linux ]
    + [ ( 4 -eq 3 -a 2 -ge 8 ) -o 4 -gt 3 ]
    + command_exists ip
    + command -v ip
    + ip netns list
    + command_exists nsenter
    + command -v nsenter
    + [ 1 -gt 0 ]
    + COMMAND=launch
    + shift 1
    + handle_help_arg 
    + [  = --help ]
    + deprecation_warnings
    + [ 0 -gt 0 ]
    + check_not_running router weave weaveworks/weave
    + docker inspect --format={{.State.Running}} {{.State.Status}} {{.Config.Image}} weave
    + RUN_STATUS=
    + true
    + check_not_running proxy weaveproxy weaveworks/weaveexec
    + docker inspect --format={{.State.Running}} {{.State.Status}} {{.Config.Image}} weaveproxy
    + RUN_STATUS=
    + true
    + check_not_running plugin weaveplugin weaveworks/plugin
    + docker inspect --format={{.State.Running}} {{.State.Status}} {{.Config.Image}} weaveplugin
    + RUN_STATUS=
    + true
    + common_launch_args
    + args=
    + [ 0 -gt 0 ]
    + echo 
    + COMMON_ARGS=
    + launch_router
    + LAUNCHING_ROUTER=1
    + check_forwarding_rules
    + run_iptables -C FORWARD -j REJECT --reject-with icmp-host-prohibited
    + enforce_docker_bridge_addr_assign_type
    + cat /sys/class/net/docker0/addr_assign_type
    + ADDR_ASSIGN_TYPE=3
    + [ 3 != 3 ]
    + is_cidr 
    + echo 
    + grep -E ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/[0-9]{1,2}$
    + CONTAINER_PORT=6783
    + ARGS=
    + IPRANGE=
    + IPRANGE_SPECIFIED=
    + docker_bridge_ip
    + ip -4 addr show dev docker0
    + grep -m1 -o inet [.0-9]*
    + DOCKER_BRIDGE_IP=inet 172.17.0.1
    + DOCKER_BRIDGE_IP=172.17.0.1
    + DNS_ROUTER_OPTS=--dns-listen-address 172.17.0.1:53
    + NO_DNS_OPT=
    + [ 0 -gt 0 ]
    + eval set -- 
    + set --
    + create_bridge
    + detect_bridge_type
    + BRIDGE_TYPE=
    + [ -d /sys/class/net/datapath ]
    + [ -d /sys/class/net/weave ]
    + return 1
    + BRIDGE_TYPE=bridge
    + [ -z  ]
    + BRIDGE_TYPE=bridged_fastdp
    + [ -n  ]
    + util_op create-datapath datapath
    + command_exists weaveutil
    + command -v weaveutil
    + weaveutil create-datapath datapath
    + :
    + init_bridged_fastdp
    + init_fastdp
    + MTU=1410
    + ip link set dev datapath mtu 1410
    + init_bridge
    + [ -n 1410 ]
    + ip link add name weave type bridge
    + mac_from_file /sys/class/dmi/id/product_uuid
    + cat /sys/class/dmi/id/product_uuid
    + UUID=C2B81CFF-9E01-11E5-955F-812AC805F0FF
    + sha256sum
    + cut -c1-17
    + mac_from_hex
    + read a b c d e f
    + sed s/\(..\)/\1 /g
    + echo 9oBJ0Jmip-C2B81CFF-9E01-11E5-955F-812AC805F0FF
    + printf %02x:1e:28:6d:ad:e6 158
    + MAC=9e:1e:28:6d:ad:e6
    + ip link set dev weave address 9e:1e:28:6d:ad:e6
    + ip link add name vethwedu mtu 1410 type dummy
    + ip link set dev vethwedu master weave
    + ip link del dev vethwedu
    + create_veth vethwe-bridge vethwe-datapath configure_veth_bridged_fastdp
    + VETHL=vethwe-bridge
    + VETHR=vethwe-datapath
    + shift 2
    + ip link show vethwe-bridge
    + ip link add name vethwe-bridge mtu 1410 type veth peer name vethwe-datapath mtu 1410
    + ip link set vethwe-bridge up
    + ip link set vethwe-datapath up
    + configure_veth_bridged_fastdp
    + add_iface_fastdp vethwe-datapath
    + util_op add-datapath-interface datapath vethwe-datapath
    + command_exists weaveutil
    + command -v weaveutil
    + weaveutil add-datapath-interface datapath vethwe-datapath
    + add_iface_bridge vethwe-bridge
    + ip link set vethwe-bridge master weave
    + ip link set dev datapath up
    + [ docker0 != weave ]
    + run_iptables -t filter -I FORWARD -i docker0 -o weave -j DROP
    + [ -z 1 ]
    + iptables -w -t filter -I FORWARD -i docker0 -o weave -j DROP
    + [ -n 172.17.0.1 ]
    + add_iptables_rule filter INPUT -i docker0 -p tcp --dst 172.17.0.1 --dport 6783 -j DROP
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C INPUT -i docker0 -p tcp --dst 172.17.0.1 --dport 6783 -j DROP
    + run_iptables -t filter -A INPUT -i docker0 -p tcp --dst 172.17.0.1 --dport 6783 -j DROP
    + [ -z 1 ]
    + iptables -w -t filter -A INPUT -i docker0 -p tcp --dst 172.17.0.1 --dport 6783 -j DROP
    + add_iptables_rule filter INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6783 -j DROP
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6783 -j DROP
    + run_iptables -t filter -A INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6783 -j DROP
    + [ -z 1 ]
    + iptables -w -t filter -A INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6783 -j DROP
    + add_iptables_rule filter INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6784 -j DROP
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6784 -j DROP
    + run_iptables -t filter -A INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6784 -j DROP
    + [ -z 1 ]
    + iptables -w -t filter -A INPUT -i docker0 -p udp --dst 172.17.0.1 --dport 6784 -j DROP
    + add_iptables_rule filter INPUT -i docker0 -p udp --dport 53 -j ACCEPT
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C INPUT -i docker0 -p udp --dport 53 -j ACCEPT
    + run_iptables -t filter -A INPUT -i docker0 -p udp --dport 53 -j ACCEPT
    + [ -z 1 ]
    + iptables -w -t filter -A INPUT -i docker0 -p udp --dport 53 -j ACCEPT
    + add_iptables_rule filter INPUT -i docker0 -p tcp --dport 53 -j ACCEPT
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C INPUT -i docker0 -p tcp --dport 53 -j ACCEPT
    + run_iptables -t filter -A INPUT -i docker0 -p tcp --dport 53 -j ACCEPT
    + [ -z 1 ]
    + iptables -w -t filter -A INPUT -i docker0 -p tcp --dport 53 -j ACCEPT
    + add_iptables_rule filter FORWARD -i weave -o weave -j ACCEPT
    + IPTABLES_TABLE=filter
    + shift 1
    + run_iptables -t filter -C FORWARD -i weave -o weave -j ACCEPT
    + run_iptables -t filter -A FORWARD -i weave -o weave -j ACCEPT
    + [ -z 1 ]
    + iptables -w -t filter -A FORWARD -i weave -o weave -j ACCEPT
    + run_iptables -t nat -N WEAVE
    + add_iptables_rule nat POSTROUTING -j WEAVE
    + IPTABLES_TABLE=nat
    + shift 1
    + run_iptables -t nat -C POSTROUTING -j WEAVE
    + run_iptables -t nat -A POSTROUTING -j WEAVE
    + [ -z 1 ]
    + iptables -w -t nat -A POSTROUTING -j WEAVE
    + [  = --without-ethtool -o -n  ]
    + ethtool_tx_off_bridged_fastdp weave
    + true
    + ip link set dev weave up
    + configure_arp_cache weave
    + sh -c echo 5 >/proc/sys/net/ipv4/neigh/weave/base_reachable_time &&
                  echo 2 >/proc/sys/net/ipv4/neigh/weave/delay_first_probe_time &&
                  echo 1 >/proc/sys/net/ipv4/neigh/weave/ucast_solicit
    + cat /sys/class/net/weave/address
    + PEERNAME=9e:1e:28:6d:ad:e6
    + [ -z  ]
    + IPRANGE=10.32.0.0/12
    + check_overlap 10.32.0.0/12
    + util_op netcheck 10.32.0.0/12 weave
    + command_exists weaveutil
    + command -v weaveutil
    + weaveutil netcheck 10.32.0.0/12 weave
    + docker inspect -f   weavedb
    + protect_against_docker_hang
    + check_running weaveplugin
    + rm -f /run/docker/plugins/weave.sock /run/docker/plugins/weavemesh.sock
    + docker create -v /weavedb --name=weavedb --label=weavevolumes weaveworks/weavedb


After that I have:


    osallou@tifenn~/Development/NOSAVE/go-docker/weave $ sudo docker ps
    CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS                  PORTS               NAMES
    c978be69b249        weaveworks/weave:1.6.0   "/home/weave/weaver -"   4 seconds ago       Up Less than a second                       weave


No weavevolumes running or launched/failed:

    osallou@tifenn~/Development/NOSAVE/go-docker/weave $ sudo docker inspect weavevolumes-1.6.0
    []
    Error: No such image, container or task: weavevolumes-1.6.0

**Comment by rade** [comment]
Looks like for some reason docker is not pulling the required images from dockerhub.

Try running `weave setup` and post the output here.

**Comment by osallou** [comment]
osallou@tifenn~/Development/NOSAVE/go-docker/weave $ sudo ./weave setup
    1.6.0: Pulling from weaveworks/weave
    Digest: sha256:681ab7e42eaf554b2246140b90b4bfb97a7c8554605f5f9cdad2b2aad265f7ad
    Status: Image is up to date for weaveworks/weave:1.6.0
    1.6.0: Pulling from weaveworks/weaveexec
    Digest: sha256:3da2ef965b86afdb5a2e597cda370e8381cdc09d9905704ebf7b59ebf830b570
    Status: Image is up to date for weaveworks/weaveexec:1.6.0
    1.6.0: Pulling from weaveworks/plugin
    Digest: sha256:f426d4ea4def6a1a56bc95ae46b1b0635c5cb5ac031c4bf7053143b7e9158104
    Status: Image is up to date for weaveworks/plugin:1.6.0
    latest: Pulling from weaveworks/weavedb
    Digest: sha256:fcc8f2cd41bb6cf5d8649f29512254b9b6b1d907bfe268b81875577edf285c42
    Status: Image is up to date for weaveworks/weavedb:latest

**Comment by rade** [comment]
Hmm. Are you sure that trace from https://github.com/weaveworks/weave/issues/2483#issuecomment-241636889 is complete? It finishes before starting the router, yet you have a running router afterwards.

**Comment by osallou** [comment]
Yes, I even retried, logging stops at:

    ...
    + docker inspect -f   weavevolumes-1.6.0
    + protect_against_docker_hang
    + check_running weaveplugin
    + rm -f /run/docker/plugins/weave.sock /run/docker/plugins/weavemesh.sock
    + docker create -v /w -v /w-noop -v /w-nomcast --name=weavevolumes-1.6.0 --label=weavevolumes --entrypoint=/bin/false weaveworks/weaveexec:1.6.0
     osallou@tifenn~/Development/NOSAVE/go-docker/weave $ sudo docker ps
     CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS               NAMES
     f3a81cf720e8        weaveworks/weave:1.6.0   "/home/weave/weaver -"   17 seconds ago      Up 3 seconds                            weave

**Comment by brb** [comment]
@osallou what exit code (`echo $?`) do you get after `weave launch`?

**Comment by osallou** [comment]
Exit code 1

**Comment by brb** [comment]
Thanks. Anything suspicious in the system logs (`journalctl -r -u docker`, if you are using systemd)?

**Comment by osallou** [comment]
There are indeed strange logs:

    -- Logs begin at Tue 2016-08-23 07:24:40 CEST, end at Tue 2016-08-23 12:00:09 CEST. --
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.576659957+02:00" level=info msg="API listen on /var/run/docker.sock"
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.576610691+02:00" level=info msg="API listen on 127.0.0.1:2375"
    Aug 23 12:00:09 tifenn systemd[1]: Started Docker Application Container Engine.
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.572144895+02:00" level=info msg="Docker daemon" commit=8eab29e graphdriver=aufs version=1.12.0
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.572130867+02:00" level=info msg="Daemon has completed initialization"
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.572081973+02:00" level=info msg="Loading containers: done."
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.330133555+02:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP a
    Aug 23 12:00:09 tifenn dockerd[27949]: time="2016-08-23T12:00:09.269842319+02:00" level=info msg="Removing stale sandbox efb8965ca9f5112ceb96f62d992cdb2e8ac93f5daf5197141c03dbc4c44360d6 (fd3628e2f1f02a4efaf80a4d118681327dc3c
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.890781847+02:00" level=info msg="Removing stale sandbox 682b0c571e3786264e20d9786a6cfa08b30a1ab2baa4e98271b3d5be0de18424 (bc090371b61fed2958ca63586a93f813deccd
    Aug 23 12:00:08 tifenn dockerd[27949]: .........................time="2016-08-23T12:00:08.780132913+02:00" level=info msg="Firewalld running: false"
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.669067083+02:00" level=info msg="Loading containers: start."
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.668859545+02:00" level=warning msg="mountpoint for pids not found"
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.668776413+02:00" level=warning msg="Your kernel does not support swap memory limit."
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.668581724+02:00" level=info msg="Graph migration to content-addressability took 0.00 seconds"
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.476801321+02:00" level=info msg="[graphdriver] using prior storage driver \"aufs\""
    Aug 23 12:00:08 tifenn dockerd[27949]: time="2016-08-23T12:00:08.447329810+02:00" level=warning msg="libcontainerd: unknown container bc090371b61fed2958ca63586a93f813deccd80b74f6df344a0607ea3403e610"
    Aug 23 12:00:07 tifenn dockerd[27949]: time="2016-08-23T12:00:07.445929192+02:00" level=info msg="libcontainerd: new containerd process, pid: 27955"
    Aug 23 12:00:07 tifenn dockerd[27949]: time="2016-08-23T12:00:07.445385576+02:00" level=warning msg="[!] DON'T BIND ON ANY IP ADDRESS WITHOUT setting -tlsverify IF YOU DON'T KNOW WHAT YOU'RE DOING [!]"
    Aug 23 12:00:07 tifenn systemd[1]: Starting Docker Application Container Engine...
    Aug 23 12:00:07 tifenn systemd[1]: docker.service: Failed with result 'exit-code'.
    Aug 23 12:00:07 tifenn systemd[1]: docker.service: Unit entered failed state.
    Aug 23 12:00:07 tifenn systemd[1]: docker.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/plugins/plugins.go:257 +0x5a5
    Aug 23 12:00:07 tifenn dockerd[24580]: created by github.com/docker/docker/pkg/plugins.GetAll
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/plugins/plugins.go:255 +0x5f
    Aug 23 12:00:07 tifenn dockerd[24580]: github.com/docker/docker/pkg/plugins.GetAll.func1(0xc820e57260, 0xc820e6dce0, 0xc820e5b874, 0xa)
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/plugins/plugins.go:172 +0xa2
    Aug 23 12:00:07 tifenn dockerd[24580]: github.com/docker/docker/pkg/plugins.loadWithRetry(0xc820e5b874, 0xa, 0xc820e57200, 0xc820b4e080, 0x0, 0x0)
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/plugins/discovery.go:80 +0xa65
    Aug 23 12:00:07 tifenn dockerd[24580]: github.com/docker/docker/pkg/plugins.(*localRegistry).Plugin(0xc820043e70, 0xc820e5b874, 0xa, 0x0, 0x0, 0x0)
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/plugins/discovery.go:119 +0x221
    Aug 23 12:00:07 tifenn dockerd[24580]: github.com/docker/docker/pkg/plugins.readPluginJSONInfo(0xc820e5b874, 0xa, 0xc820e5bb60, 0x23, 0x0, 0x0, 0x0)
    Aug 23 12:00:07 tifenn dockerd[24580]: /usr/local/go/src/runtime/panic.go:481 +0x3e6
    Aug 23 12:00:07 tifenn dockerd[24580]: panic(0x1a750e0, 0xc82000e090)
    Aug 23 12:00:07 tifenn dockerd[24580]: goroutine 406 [running]:
    Aug 23 12:00:07 tifenn dockerd[24580]: [signal 0xb code=0x1 addr=0x8 pc=0xc50151]
    Aug 23 12:00:07 tifenn dockerd[24580]: panic: runtime error: invalid memory address or nil pointer dereference
    Aug 23 12:00:06 tifenn dockerd[24580]: time="2016-08-23T12:00:06.885293822+02:00" level=error msg="Handler for GET /v1.18/images/weaveplugin/json returned error: No such image: weaveplugin"
    Aug 23 12:00:06 tifenn dockerd[24580]: time="2016-08-23T12:00:06.884714017+02:00" level=error msg="Handler for GET /v1.18/containers/weaveplugin/json returned error: No such container: weaveplugin"

**Comment by brb** [comment]
From the log it seems to be a Docker bug. Could you file an issue at https://github.com/docker/docker? Thanks.

**Comment by osallou** [comment]
Just upgraded to Docker 1.12.1 and it seems to work.

**Comment by brb** [comment]
Thanks for letting us know.

**Comment by fabiofumarola** [comment]
I got the same error

```
INFO: 2016/10/23 17:34:07.027126 weave proxy 1.7.2
INFO: 2016/10/23 17:34:07.027167 Command line arguments: -H unix:///var/run/weave/weave.sock
INFO: 2016/10/23 17:34:07.028215 Docker API on unix:///var/run/docker.sock: &[Arch=amd64 KernelVersion=4.4.0-45-generic BuildTime=Tue, 27 Sep 2016 12:25:38 +1300 Version=1.12.1 ApiVersion=1.24 GitCommit=23cf638 GoVersion=go1.6.2 Os=linux]
FATA: 2016/10/23 17:34:07.028359 Could not start proxy: Could not obtain address of docker0
```

Note that I changed the default directory for docker from `/var/lib/docker` to `/data/var/lib/docker/` adding a symlink to the original folder

**Comment by brb** [comment]
@fabiofumarola The error you get is not the same: "Could not start proxy: **Could not obtain address of docker0**". Could you create a new issue and include an output of the `ip -o -4 a docker0` cmd run on a host running docker / weave? Thanks.
